### PR TITLE
update cudf serialization/deserialization

### PR DESCRIPTION
**Pull request by quasiben**
This in conjunction with:
- https://github.com/rapidsai/cudf/pull/2378
- https://github.com/rapidsai/ucx-py/pull/129

Will fix: https://github.com/dask/distributed/issues/2830

I would not merge this until https://github.com/rapidsai/cudf/pull/2378 is merged in

### CODE REVIEW

File: distributed/__init__.py [hunk 1]
```diff
@@ -35,3 +35,19 @@
...
+def f(idx):
+    while True:
+        time.sleep(1)
+        frame = sys._current_frames()[idx]
+        print("-" * 100)
+        print("".join(call_stack(frame)))
+
+
+thread = threading.Thread(target=f, args=(threading.get_ident(),), daemon=True)
+# thread.start()
```
Comment (mrocklin) [inline distributed/__init__.py]: We'll want to clear this out

File: distributed/protocol/cudf.py [hunk 2]
```diff
@@ -1,74 +1,20 @@
...
-        "is_cuda": len(arrays),
-        "subheaders": sub_headers,
-        # TODO: the header must be msgpack (de)serializable.
-        # See if we can avoid names, and just use integer positions.
-        "columns": x.columns.tolist(),
-        "null_counts": null_counts,
-        "null_subheaders": null_headers,
-    }
+    header, frames = x.serialize()
+    header["is_cuda"] = 1
```
Comment (mrocklin) [inline distributed/protocol/cudf.py]: Do we need this?  I can see that it is added in all of cudf, cupy, numba, but I'm not sure that it is actually used anywhere.

We an address that in a follow-on PR though.

**Comment by quasiben** [comment]
@mrocklin I think this is ready to be merged in.  There are still issues with running a distributed join with cudf+ucx+dask.  Folks are investigating ucx-py and cudf and I think it best to get everyone on "master"

**Comment by mrocklin** [comment]
OK, there is a dask release happening later today.  Should this go in as part of that release (is it critical?) or can we merge it after the release occurs?  I have a preference for afterwards, just so that it gets a bit more use by bleeding edge users before it goes into a proper release.

**Comment by quasiben** [comment]
Afterwards makes sense.  Thanks!

**Pull request merged**